docs/log_format.md

```diff
@@ -0,0 +1,58 @@
+# Log Format
+
+## Log Line Prefix
+
+Shadow produces simulator log messages in the following format:
+
+```text
+real-time [thread-id] virtual-time [logdomain-loglevel] [hostname~ip] [function-name] MESSAGE
+```
+
+**FIXME**: The `real-time` and `virtual-time` use different time units. Is this correct, or should we change this in Shadow?
+
+- `real-time`:  
+the wall clock time since the start of the experiment, represented as `hours:minutes:seconds:microseconds`
+- `thread-id`:  
+the ID of the worker thread that generated the message
+- `virtual-time`:  
+the simulated time since the start of the experiment, represented as `hours:minutes:seconds:nanoseconds`
+- `logdomain`:  
```

Comment:
We don't currently log such a field.

I *had* been thinking it might be useful to log contents of plugin writes to stdout, etc., in which case we could use such a field. OTOH it could also just be part of the log message itself.

OTOOH I had also been thinking it might make sense to log the current process and thread when applicable, though that might make more sense in the next field. e.g. `[hostname~ip.plugin-name.thread-id]`.

Assuming we don't want to hold up this change on new features, probably makes sense to just drop this field from the docs for now.

Reply:
I removed this field, and also noticed the documentation was missing the `[file:line]` field so I added that to the documentation.

> I had been thinking it might be useful to log contents of plugin writes to stdout, etc., in which case we could use such a field. OTOH it could also just be part of the log message itself.

Do you mean logging shim messages to Shadow's log rather than the separate shim log file?

> OTOOH I had also been thinking it might make sense to log the current process and thread when applicable, though that might make more sense in the next field. e.g. [hostname~ip.plugin-name.thread-id].

Logging the plugin name and thread sounds like a good idea to me. A downside might be that the log lines are already very long and would get longer, but I think having more information is better than less.